pandas/core/indexes/interval.py

```diff
@@ -546,7 +546,7 @@ def _maybe_convert_i8(self, key):
             if lib.is_period(key):
                 key_i8 = key.ordinal
             elif isinstance(key_i8, Timestamp):
-                key_i8 = key_i8.value
+                key_i8 = key_i8._value
```

Comment:
looks like 1) this scalar path might not be necessary? and 2) there may be baked-in assumptions about everything being nano

Reply:
good call - shall we make logic changes in a separate PR? I've opened https://github.com/pandas-dev/pandas/issues/51196 about this nanosecond assumption here

Reply:
separate PR seems fine

Reply:
sure, thanks - any objections here? shall we move forward before merge conflicts arise?